Test/git-issues/git-issue-4202.dfy

```diff
@@ -0,0 +1,15 @@
+// RUN: %baredafny verify %args "%s" > "%t"
+// RUN: %diff "%s.expect" "%t"
+
+opaque function F(x: int): char
+{ 'D' }
+
+method InitArray<D>(f: int -> D) returns (a: D)
+{
+  return f(44);
+}
+
+method Main() {
+  reveal F();
+  var c := InitArray(F);
```

Comment:
```suggestion
  var c := InitArray(F);
  assert c == 'D';
```
This would help ensure not only that it doesn't crash but also that `F` is properly revealed.